File: core/repository/intelligence/build_system_engine.py
from __future__ import annotations

def detect_build_systems(text: str):
    src=(text or '').lower()
    out=[]
    for k,v in [('pyproject.toml','python'),('package.json','node'),('pom.xml','maven'),('build.gradle','gradle'),('cargo.toml','cargo')]:
        if k in src: out.append(v)
    return sorted(set(out))

File: src/repository/intelligence/buildSystemEngine.ts
/**
 * Converted from Python: core/repository/intelligence/build_system_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../../runtime/pyCompat.js";

export function detectBuildSystems(text: any): any {
  var src: any = String(py.or2(text, () => (""))).toLowerCase();
  var out: any[] = [];
  var k: any;
  var v: any;
  for ([k, v] of py.iter([["pyproject.toml", "python"], ["package.json", "node"], ["pom.xml", "maven"], ["build.gradle", "gradle"], ["cargo.toml", "cargo"]])) {
    if (py.contains(src, k)) {
      py.listAppend(out, v);
    }
  }
  return py.sorted(py.toSet(out));
}
